### PR TITLE
nanotts: init at 2021-2-22

### PR DESCRIPTION
**Pull request by strikerlulu**
Add [nanotts](https://github.com/gmn/nanotts) which is an Improved SVOX PicoTTS speech synthesizer
using it is easy as `echo "testing" | nanotts -p` just like **espeak**
also add myself to maintainers

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by strikerlulu** [comment]
@Artturin ? :)

File: pkgs/tools/audio/nanotts/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  version = "unstable-2021-2-22";
+
+  src = fetchFromGitHub {
+    repo = "nanotts";
+    owner = "gmn";
+    rev = "d8b91f3d9d524c30f6fe8098ea7a0a638c889cf9";
+    sha256 = "sha256-bFu3U50zc90iQeWkqOsCipkueJUZI3cW5342jjYSnGI=";
+  };
+  nativeBuildInputs = [ autoconf automake ];
+  buildInputs = [ libtool popt alsaLib ];
```
Comment (Artturin) [inline pkgs/tools/audio/nanotts/default.nix:15]:
```suggestion
  strictDeps = true;
  nativeBuildInputs = [ autoconf automake libtool ];
  buildInputs = [ popt alsaLib ];
```

libtool is run during the build so it should be in nativeBuildInputs

strictDeps ensures that dependencies are correctly declared (its not the default yet sadly)

File: pkgs/tools/audio/nanotts/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  patchPhase = ''
+  substituteInPlace "src/main.cpp" --replace "/usr/share/pico/lang" "$out/share/lang"
+  echo "" > update_build_version.sh
+  '';
+  installPhase = ''
+    mkdir -p $out/bin
+    install -m 0755 nanotts $out/bin/
+    mkdir $out/share
+    cp -r $src/lang $out/share/lang
```
Comment (Artturin) [inline pkgs/tools/audio/nanotts/default.nix:24]:
```suggestion
    install -Dm644 -t $out/bin nanotts
    install -Dm644 -t $out/share lang
```

**Review by Artturin — CHANGES_REQUESTED** [review]
please use `nixpkgs-fmt` on the file and run `nix shell -f https://github.com/jtojnar/nixpkgs-hammering/archive/master.tar.gz -c nixpkgs-hammer nanotts` it'll give you some tips

File: pkgs/tools/audio/nanotts/default.nix (hunk 2 above)
Comment (strikerlulu) [reply pkgs/tools/audio/nanotts/default.nix:24]: shouldn't binary have `755` permission mode ?
or you were just suggesting to use the `install` command

Comment (Artturin) [reply pkgs/tools/audio/nanotts/default.nix]: i edited the suggestion

Comment (strikerlulu) [reply pkgs/tools/audio/nanotts/default.nix]: I've already pushed with that suggestion :)

**Comment by Artturin** [comment]
the commits should be amended to the init commit

File: pkgs/tools/audio/nanotts/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  meta = {
+    description = "Speech synthesizer commandline utility that improves pico2wave, included with SVOX PicoTTS";
+    homepage = "https://github.com/gmn/nanotts";
+    license = lib.licenses.asl20;
+    maintainers = [ lib.maintainers.strikerlulu ];
+    platforms = lib.platforms.linux;
+  };
+}
+
+
```
Comment (Artturin) [inline pkgs/tools/audio/nanotts/default.nix:33]:
```suggestion
```

File: pkgs/tools/audio/nanotts/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    repo = "nanotts";
+    owner = "gmn";
+    rev = "d8b91f3d9d524c30f6fe8098ea7a0a638c889cf9";
+    sha256 = "sha256-bFu3U50zc90iQeWkqOsCipkueJUZI3cW5342jjYSnGI=";
+  };
+  strictDeps = true;
+  nativeBuildInputs = [ autoconf automake libtool ];
+  buildInputs = [ popt alsaLib ];
```
Comment (Artturin) [inline pkgs/tools/audio/nanotts/default.nix:15]:
```suggestion
  buildInputs = [ popt alsaLib ];

```

File: pkgs/tools/audio/nanotts/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  patchPhase = ''
+    substituteInPlace "src/main.cpp" --replace "/usr/share/pico/lang" "$out/share/lang"
+    echo "" > update_build_version.sh
+  '';
+  installPhase = ''
+    install -Dm755 -t $out/bin nanotts
+    install -Dm644 -t $out/share/lang $src/lang/*
+  '';
```
Comment (Artturin) [inline pkgs/tools/audio/nanotts/default.nix:23]:
```suggestion
  '';

```

File: pkgs/tools/audio/nanotts/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    rev = "d8b91f3d9d524c30f6fe8098ea7a0a638c889cf9";
+    sha256 = "sha256-bFu3U50zc90iQeWkqOsCipkueJUZI3cW5342jjYSnGI=";
+  };
+  strictDeps = true;
+  nativeBuildInputs = [ autoconf automake libtool ];
+  buildInputs = [ popt alsaLib ];
+  patchPhase = ''
+    substituteInPlace "src/main.cpp" --replace "/usr/share/pico/lang" "$out/share/lang"
+    echo "" > update_build_version.sh
+  '';
```
Comment (Artturin) [inline pkgs/tools/audio/nanotts/default.nix:19]:
```suggestion
  '';

```

**Pull request merged**